### ISSUE TITLE
Mobile Menu: Clicking a link doesn't close the menu. 

### ISSUE THREAD

**Issue by joseph-abell** [hannahf86/portfolio#1]
Hey, firstly I wanted to say, good work on your website. Genuinely looks good, don't take a look in wayback machine on my website, whatever you do.

Let me know if you want issues like this raising, or if you've got enough on. 

When viewing your site on android, Firefox, I noticed that there wasn't a way to close your mobile Menu other than clicking the homepage link. I'd have expected other links to close the menu, and for clicking the overlay that covers the site to do the same.

Cheers, and once again, well done.

Joe

**Comment by hannahf86** [comment]
Hi Joe,

Thanks very much! I was aware but haven't got to it yet. Mainly because I have stared at that code for so long and I have no idea why it isn't working. I might have a look now actually. Currently I'm trying to sort out a fetch and post method for a newsletter and I'm pretty sure I've over complicated it!

I'm not actually sure why the menu isn't closing tbh. This was the first project I did in React after the learning part of the course and I'm looking at it now and I ask myself "... why...?" for so many things. But I genuinely can't see why the menu closes on the Home button but not the others. Do you have any idea? All I can think is that its perhaps because I haven't specifically told it to disappear. And actually, its more the Tailwind layout that upsets me. There's no logic to the order!

I'm mid tidying my github atm. Firstly I changed my username because it was my married name, which was fun, and then I cleared it of all the small projects from the course. I need to write up my README files, actually load the code back up for some projects I haven't done yet. But, it is all tidy now and looks more organised I think.

Any help you can offer will be HUGELY appreciated. You have to learn 10+ languages in 7 months so there wasn't a lot of time to make sure it really sticks so I do have some blind spots with some things. I understand it all, obviously, hence passing the exams, but sometimes I feel like I'm completely lost!

Sorry for the rant!

Thanks again,
Hannah

**Comment by joseph-abell** [comment]
so the difference between the home link and the other links is that the home link points to a url, the other links point to an id. If you changed home to point to `#` the functionality would be consistent throughout, which is a good start.

Because you want to add functionality to links before navigating to the element, I'd suggest changing the links to buttons, and using their onClick handle to use handleNav. You might need to pass it a second argument for the element you want to navigate to, and use the react-router-dom api to do the navigation programatically. I'm happy to go into more detil if needed.

**Comment by joseph-abell** [comment]
I've played with tailwind a fair few times, and although I see it's utility in quickly building up a cohesive style, I find the cruft classNames littered everywhere to be a nuisance. Maybe it's because I started as a CSS developer, but I much prefer to craft my styling by hand.

**Comment by hannahf86** [comment]
Yeah, Tailwind is good for basic, quick things, but real CSS is far superior these days. I follow a chap called Kevin Powell on YouTube and he is a CSS legend.

**Comment by hannahf86** [comment]
> so the difference between the home link and the other links is that the home link points to a url, the other links point to an id. If you changed home to point to `#` the functionality would be consistent throughout, which is a good start.
> 
> Because you want to add functionality to links before navigating to the element, I'd suggest changing the links to buttons, and using their onClick handle to use handleNav. You might need to pass it a second argument for the element you want to navigate to, and use the react-router-dom api to do the navigation programatically. I'm happy to go into more detil if needed.

I was looking at it this morning and I thought to myself there's no way I'd write it out like that now. If I was to remake that nav system I think I would have made them buttons. I'll have a go at it first and let you know if I get stuck!

Thanks very much Joe!
